src/main/java/org/agbrothers/englishbot/processing/ProcessingCore.java

```diff
@@ -23,62 +24,70 @@
 public class ProcessingCore {
     private TelegramBot telegramBot;
     private LessonService lessonService;
-    private final Map<String, String> stateByChatId = new HashMap<>();
-    private ProcessorHolder processorHolder;
+    private ProcessorFactory processorFactory;
+    private UserService userService;
 
     public void processUserRequest(String chatId, String messageText) {
-        String stateId = getStateId(chatId);
-        if (stateId == null) {
-            sendHelloMessage(chatId);
-            stateByChatId.put(chatId, State.MAIN_MENU);
+        User user = userService.getUserByChatId(chatId);
```

Comment:
Actually I liked that you used ternar operator in previous commit. Why did you changed the approach?